lib/analytics/widgets/analytics_cart_listener.dart

```diff
@@ -0,0 +1,64 @@
+import 'dart:async';
+
+import 'package:flex_storefront/analytics/apis/analytics_api.dart';
+import 'package:flex_storefront/analytics/models/analytics_events.dart';
+import 'package:flex_storefront/cart/cart_repository.dart';
+import 'package:flex_storefront/cart/models/cart_message.dart';
+import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
+
+class CartMessageListener extends StatefulWidget {
+  final Widget child;
+
+  const CartMessageListener({
+    super.key,
+    required this.child,
+  });
+
+  @override
+  State<CartMessageListener> createState() => _CartMessageListenerState();
+}
+
+class _CartMessageListenerState extends State<CartMessageListener> {
+  late final StreamSubscription<CartMessage> _subscription;
+
+  @override
+  void initState() {
+    super.initState();
+    _subscription = GetIt.instance
+        .get<CartRepository>()
+        .getCartMessageStream()
+        .listen(_onMessage);
+  }
+
+  void _onMessage(CartMessage event) {
+    if (event is ChangeQuantityMessage) {
+      if (event.quantityDiff >= 0) {
```

Comment:
I think we can have a `quantityDiff` of -1 meaning, "decrease quantity by 1".. but it isn't a RemoveFromCart.

Or maybe I'm reading this wrong?

Reply:
Yep, so as discussed yesterday, I have refactor it to not tracking the quantity updates, we are just tracking cart entry additions or removals.